test/controllers/api/admin/users_controller_test.rb

```diff
@@ -11,63 +11,313 @@ class Api::Admin::UsersControllerTest < ActionController::TestCase
-    alice = users(:alice)
-    login_as(:bob)
+        bob.reload
+        assert bob.admin?
+        assert bob.teammembers.find_by(team_id: teams(:team1))
+      end
+      
+      test 'admin updates conf_admin to admin' do
+        teammembers(:team1_bob).destroy
+        conf_admin = users(:conf_admin)
 
-    delete :destroy, params: { id: alice.id }
+        login_as(:admin)
+        patch :update_role, params: { user_id: conf_admin, role: :admin }, xhr: true
 
-    assert alice.reload.persisted?
-    assert_includes(errors, 'Access denied')
-    assert_equal 403, status_code
-  end
+        conf_admin.reload
+        assert conf_admin.admin?
+        assert conf_admin.teammembers.find_by(team_id: teams(:team1))
+      end
+
+      test 'admin updates admin to conf_admin' do
+        teammembers(:team1_bob).destroy
+        admin2 = Fabricate(:admin)
+
+        login_as(:admin)
+        patch :update_role, params: { user_id: admin2, role: :conf_admin }, xhr: true
+
+        admin2.reload
+        assert admin2.conf_admin?
+        assert_not admin2.teammembers.find_by(team_id: teams(:team1))
+      end
+      
+      test 'admin updates admin to user' do
+        teammembers(:team1_bob).destroy
+        admin2 = Fabricate(:admin)
+
+        login_as(:admin)
+        patch :update_role, params: { user_id: admin2, role: :user }, xhr: true
+
+        admin2.reload
+        assert admin2.user?
+        assert_not admin2.teammembers.find_by(team_id: teams(:team1))
+      end
+      
+      test 'admin updates conf_admin to user' do
+        teammembers(:team1_bob).destroy
+        conf_admin = users(:conf_admin)
+
+        login_as(:admin)
+        patch :update_role, params: { user_id: conf_admin, role: :user }, xhr: true
+
+        conf_admin.reload
+        assert conf_admin.user?
+        assert_not conf_admin.teammembers.find_by(team_id: teams(:team1))
+      end
+      
+      test 'admin cannot update himself' do
+        teammembers(:team1_bob).destroy
+        admin = users(:admin)
+
+        login_as(:admin)
+        patch :update_role, params: { user_id: admin, role: :conf_user }, xhr: true
+
+        admin.reload
+        assert admin.admin?
+        assert admin.teammembers.find_by(team_id: teams(:team1))
+      end
+    end
+    
+    context 'conf_admin' do
+      test 'conf admin updates user to conf admin' do
+        teammembers(:team1_bob).destroy
+        bob = users(:bob)
+
+        login_as(:tux)
+        patch :update_role, params: { user_id: bob, role: :conf_admin }, xhr: true
+
+        bob.reload
+        assert bob.conf_admin?
+        assert_not bob.teammembers.find_by(team_id: teams(:team1))
+      end
+
+      test 'conf admin cannot update user to admin' do
+        teammembers(:team1_bob).destroy
+        bob = users(:bob)
+
+        login_as(:tux)
+        patch :update_role, params: { user_id: bob, role: :admin }, xhr: true
+
+        bob.reload
+        assert bob.user?
```

Comment:
use assert_equals true, bob.user?
or check api doc for assert

Reply:
Done